### PR TITLE
Support Cmake as a build system

### PR DESCRIPTION
**Pull request by masatake**
This is mostly based on @ParticleG's work. See #51.

### CODE REVIEW

**Comment by masatake** [comment]
@ParticleG You wrote in #51: 
> I'm coding on Windows and use [vcpkg](https://github.com/microsoft/vcpkg) to manage CMake packages.

I would like to know more about this.
Do you mean you will use libreadtags with vcpkg when I merge this pull request? If so, I want to test this use case on GitHub actions (or Appveryor?).

File: README.md [hunk 1]
```diff
@@ -10,3 +12,58 @@ NEWS.md describes note worty changes.
...
 you will like [readtags command](https://docs.ctags.io/en/latest/man/readtags.1.html)
 shipped as part of Universal-ctags. It uses libreadtags extensively.
+
+You can build libreadtags with GNU Autotools and CMake.
+
+## Autotools Usage
+
+### Build
+```shell
+./autogen.sh
```
Comment (masatake) [inline README.md]: This step is for the source code cloned from the git repository.

**Comment by ParticleG** [comment]
> @ParticleG You wrote in #51:
> 
> > I'm coding on Windows and use [vcpkg](https://github.com/microsoft/vcpkg) to manage CMake packages.
> 
> I would like to know more about this. Do you mean you will use libreadtags with vcpkg when I merge this pull request? If so, I want to test this use case on GitHub actions (or Appveryor?).

Hmmm, currently I wrote a custom `fetch_libreadtags.cmake`, which is using my forked repo, the logic is exactly the same as I mentioned in README.md: 
``` cmake
include(FetchContent)

if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/cmake/libreadtags-master.zip")
    FetchContent_Declare(
            libreadtags
            URL file://${CMAKE_CURRENT_SOURCE_DIR}/cmake/libreadtags-master.zip
    )
else ()
    FetchContent_Declare(
            libreadtags
            GIT_REPOSITORY https://github.com/ParticleG/libreadtags.git
            GIT_TAG master
    )
endif ()

FetchContent_MakeAvailable(libreadtags)
```

If you want to use this ***IN*** `vcpkg` (which is a C++ CMake based dependency manager by Microsoft), you need to create a PR in their repo and add some more stuff like manifest (Yes, it's Git-based version controll)

**Comment by ParticleG** [comment]
Also CMake come along with a testing tool called [`CTest`](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html). I'm not sure if you'd like to use it, I just pasted the link to the official documentation here

**Comment by masatake** [comment]
@ParticleG thank you for the information.

I'm interested in registering libreadtags to vckg. However, I cannot put it as a high-priority item in my queue.

Using CTest is more interesting. Anyway, I will merge this.

**Comment by masatake** [comment]
@ParticleG, thank you very much.

I'm thinking about releasing a new version.

**Comment by ParticleG** [comment]
> @ParticleG thank you for the information.
> 
> I'm interested in registering libreadtags to vckg. However, I cannot put it as a high-priority item in my queue.
> 
> Using CTest is more interesting. Anyway, I will merge this. 

I can try making a port and create a PR to vcpkg, if you want. But I'm quite busy this month, so this could go slowly

**Pull request merged**